dipy/nn/tests/test_cnn_1denoiser.py

```diff
@@ -1,18 +1,19 @@
 import pytest
-import numpy as np
 from dipy.utils.optpkg import optional_package
+from dipy.testing.decorators import set_random_number_generator
 tf, have_tf, _ = optional_package('tensorflow')
 if have_tf:
     from dipy.nn.cnn_1d_denoising import Cnn1DDenoiser
 
 
 @pytest.mark.skipif(not have_tf, reason='Requires TensorFlow')
```

Comment:
can you update here `not have_tf or not have_sklearn` 

you will have to import scikit-lean  on the top with `optional_package`.